library/hls/src/main/java/com/google/android/exoplayer2/source/hls/HlsSampleStreamWrapper.java

```diff
@@ -753,13 +837,23 @@ public void onLoadCanceled(
         loadable.startTimeUs,
         loadable.endTimeUs);
     if (!released) {
-      resetSampleQueues();
+      if (interrupted) {
+        interrupted = false;
+      }
+      else {
+        resetSampleQueues();
+      }
       if (enabledTrackGroupCount > 0) {
         callback.onContinueLoadingRequested(this);
       }
     }
   }
 
+  private void discardAfterInterrupt() {
+    int preferredQueueSize = mediaChunks.size() - 1;
```

Comment:
Might be better to save the value that was returned from chunkSource.getPreferredQueueSize and use it here.